vms/rpcchainvm/vm_server.go

```diff
@@ -387,9 +388,19 @@ func (vm *VMServer) BuildBlock(ctx context.Context, req *vmpb.BuildBlockRequest)
 	if vm.bVM == nil || req.PChainHeight == nil {
 		blk, err = vm.vm.BuildBlock(ctx)
 	} else {
-		blk, err = vm.bVM.BuildBlockWithContext(ctx, &block.Context{
+		blkContext := &block.Context{
 			PChainHeight: *req.PChainHeight,
-		})
+		}
+		switch len(req.RandomnessSeed) {
+		case 0:
+			blkContext.RandomnessSeed = [hashing.HashLen]byte{}
+		case hashing.HashLen:
+			blkContext.RandomnessSeed = [hashing.HashLen]byte(req.RandomnessSeed)
+		default:
+			// invalid input.
+			return nil, errors.New("build block request contained invalid randomness seed length")
```

Comment:
nit: usually we define a package level error instead of using rogue errors.New. It would help testing, as it allow asserting a specific error

Reply:
done